src/System.Data.Common/src/System/Data/Common/DbDataReader.cs

```diff
@@ -61,40 +65,73 @@ public virtual DataTable GetSchemaTable()
 
         public abstract bool GetBoolean(int ordinal);
 
+        public bool GetBoolean(string name) => GetBoolean(GetOrdinal(name));
+
         public abstract byte GetByte(int ordinal);
 
+        public byte GetByte(string name) => GetByte(GetOrdinal(name));
+
         public abstract long GetBytes(int ordinal, long dataOffset, byte[] buffer, int bufferOffset, int length);
 
+        public long GetBytes(string name, long dataOffset, byte[] buffer, int bufferOffset, int length) =>
+            GetBytes(GetOrdinal(name), dataOffset, buffer, bufferOffset, length);
+
         public abstract char GetChar(int ordinal);
 
+        public char GetChar(string name) => GetChar(GetOrdinal(name));
+
         public abstract long GetChars(int ordinal, long dataOffset, char[] buffer, int bufferOffset, int length);
 
+        public long GetChars(string name, long dataOffset, char[] buffer, int bufferOffset, int length) =>
+            GetChars(GetOrdinal(name), dataOffset, buffer, bufferOffset, length);
+
         [EditorBrowsable(EditorBrowsableState.Never)]
         public DbDataReader GetData(int ordinal) => GetDbDataReader(ordinal);
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public DbDataReader GetData(string name) => GetData(GetOrdinal(name));
+
         IDataReader IDataRecord.GetData(int ordinal) => GetDbDataReader(ordinal);
 
         protected virtual DbDataReader GetDbDataReader(int ordinal)
         {
             throw ADP.NotSupported();
         }
 
+        public DbDataReader GetDbDataReader(string name) => GetDbDataReader(GetOrdinal(name));
```

Comment:
This should not be implemented; delete this line.

https://github.com/dotnet/corefx/issues/31595#issuecomment-459019657

Reply:
Removed it